src/osd/scrubber/pg_scrubber.cc

```diff
@@ -937,9 +937,12 @@ bool PgScrubber::select_range()
   // debug: be 'blocked' if told so by the 'pg scrub_debug block' asok command
   if (m_debug_blockrange > 0) {
     m_debug_blockrange--;
-    return false;
+    return 0;
   }
-  return true;
+
+  if (!objects.empty())
+    return objects.size();
+  return 1;
```

Comment:
So, 0 if scrub is blocked, 1 if it's not blocked, but there are no further objects, and N otherwise?  That's pretty confusing -- how about std::optional<unsigned> where std::nullopt indicates blocked and otherwise it simply encodes the number of objects in the range?

Alternately, you could keep the return value a signed integer and use -EAGAIN to indicate that it's blocked.  @ronen-fr Do you have a preference?

Reply:
I'm definitely for the 'optional'